### ISSUE TITLE
Questions on low data classification

### ISSUE THREAD

**Issue by edward3862** [bcmi/F2GAN-Few-Shot-Image-Generation#7]
Hello, I'm interested in the low data classification experiment and its implemental detail. Could you please help me to answer my doubts?

1. I notice that you use the first 20 real images for validation and the rest 512 generated images for training in `data_for_augmentedimages_for_classifier.py`. Does that mean the real images are always used as the validation set? And how can I test the classification accuracy on the testing images of the unseen categories?

> training_dataset = self.datasets[dataset_name][:, 20:]
            testing_dataset = self.datasets[dataset_name][:, :20]

2. What does the `testing_num=10` mean in `generation_builder_with_matchingclassifier.py`?

3. It seems that the only difference between low-data experiment and few-shot experiment is the value of `selected_class` N and `support_num` K. Can one consider the low data classification as [num of unseen class]-way-1-shot setting in few-shot learning?

4. Did you use data augmentation to pre-train the ResNet 18 on the seen categories for low-data and few-shot experiments?

Thanks again :)

**Comment by hy-zpg** [comment]
1.  refer to paper K-sample setting, for all unseen categories (for example N unseen categories), using K conditional images to generate 512 images for each unseen categories. Using K*N conditional images and 512*N generated images as training images, and using remaining images of unseen categories as testing images.
2. testing_num means the number of testing images, which need to be set as different values for different datasets.
3. yes, K-sample means the [num of unseen class]-way-K-shot setting in the few-shot setting.
4. yes.

**Comment by edward3862** [comment]
> 1. refer to paper K-sample setting, for all unseen categories (for example N unseen categories), using K conditional images to generate 512 images for each unseen categories. Using K_N conditional images and 512_N generated images as training images, and using remaining images of unseen categories as testing images.

Thanks for your answer. Do you have a validation set in your experiments?
